src/Client/BuzzHouse/Generator/SQLTypes.cpp

```diff
@@ -9,143 +8,1167 @@
+        {
+            ret += next_option < 33 ? "+" : "-";
+        }
+        ret += "inf";
     }
-    else if ((it = dynamic_cast<IntType *>(tp)))
+    else if (extremes && next_option < 73)
     {
-        return new IntType(it->size, it->is_unsigned);
+        if (next_option < 65)
+        {
+            ret += next_option < 57 ? "+" : "-";
+        }
+        ret += "0.0";
     }
-    else if ((ft = dynamic_cast<FloatType *>(tp)))
+    else if (next_option < 373)
     {
-        return new FloatType(ft->size);
+        ret = std::to_string(rg.nextRandomInt32());
     }
-    else if ((dt = dynamic_cast<DateType *>(tp)))
+    else if (next_option < 673)
     {
-        return new DateType(dt->extended);
+        ret = std::to_string(rg.nextRandomInt64());
     }
-    else if ((dtt = dynamic_cast<DateTimeType *>(tp)))
+    else
     {
-        return new DateTimeType(dtt->extended, dtt->precision, dtt->timezone);
+        std::uniform_int_distribution<uint32_t> next_dist(0, 9);
+        const uint32_t left = next_dist(rg.generator);
+        const uint32_t right = next_dist(rg.generator);
+
+        ret = appendDecimal(rg, left, right);
     }
-    else if ((decp = dynamic_cast<DecimalType *>(tp)))
+    return ret;
+}
+
+String BoolType::typeName(const bool) const
+{
+    return "Bool";
+}
+
+String BoolType::MySQLtypeName(RandomGenerator &, const bool) const
+{
+    return "BOOL";
+}
+
+String BoolType::PostgreSQLtypeName(RandomGenerator &, const bool) const
+{
+    return "BOOLEAN";
+}
+
+String BoolType::SQLitetypeName(RandomGenerator &, const bool) const
+{
+    return "INTEGER";
+}
+
+SQLType * BoolType::typeDeepCopy() const
+{
+    return new BoolType();
+}
+
+String BoolType::appendRandomRawValue(RandomGenerator & rg, StatementGenerator &) const
+{
+    return rg.nextBool() ? "TRUE" : "FALSE";
+}
+
+String IntType::typeName(const bool) const
+{
+    return fmt::format("{}Int{}", is_unsigned ? "U" : "", std::to_string(size));
+}
+
+String IntType::MySQLtypeName(RandomGenerator &, const bool) const
+{
+    switch (size)
     {
-        return new DecimalType(decp->precision, decp->scale);
+        case 8:
+            return fmt::format("TINYINT {}", is_unsigned ? " UNSIGNED" : "");
+        case 16:
+            return fmt::format("SMALLINT {}", is_unsigned ? " UNSIGNED" : "");
+        case 32:
+            return fmt::format("INT {}", is_unsigned ? " UNSIGNED" : "");
+        case 64:
+            return fmt::format("BIGINT {}", is_unsigned ? " UNSIGNED" : "");
+        default:
+            assert(0);
     }
-    else if ((st = dynamic_cast<StringType *>(tp)))
+    return "";
+}
+
+String IntType::PostgreSQLtypeName(RandomGenerator &, const bool) const
+{
+    switch (size)
     {
-        return new StringType(st->precision);
+        case 8:
+        case 16:
+            return "SMALLINT";
+        case 32:
+            return "INTEGER";
+        case 64:
+            return "BIGINT";
+        default:
+            assert(0);
```

Comment:
This should probably be a `LOGICAL_ERROR`, but if you really want to use `assert` here and in other places, you'd rather use `chassert` which reports much nicer